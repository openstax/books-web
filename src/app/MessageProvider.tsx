import { shouldPolyfill } from '@formatjs/intl-pluralrules/should-polyfill';
import React from 'react';
import { IntlProvider } from 'react-intl';
import enMessages from './messages/en';

// https://formatjs.io/docs/polyfills/intl-pluralrules/#dynamic-import--capability-detection
async function polyfill(locale: 'en') {
  if (shouldPolyfill()) {
    await import('@formatjs/intl-pluralrules/polyfill');
  }

  // boolean added by the polyfill
  if ((Intl.PluralRules as (typeof Intl.PluralRules & {polyfilled?: boolean})).polyfilled) {
    switch (locale) {
      case 'en':
        await import('@formatjs/intl-pluralrules/locale-data/en');
        break;
      default:
        throw new Error('unsupported locale in messageProvider');
    }
  }
}

polyfill('en');

// tslint:disable-next-line:variable-name
const MessageProvider: React.SFC<{onError?: () => void}> = (props) =>
  <IntlProvider onError={props.onError} locale='en' messages={enMessages}>{props.children}</IntlProvider>;

export default MessageProvider;
