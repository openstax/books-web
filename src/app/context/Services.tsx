import React, { Attributes } from 'react';
import { AppServices } from '../types';

const {Consumer, Provider} = React.createContext({} as AppServices);

export {
  Consumer,
  Provider,
};

interface ServiceConsumer {
  services: AppServices;
}

/* tslint:disable-next-line:variable-name */
export default <P extends ServiceConsumer & Attributes>(Component: React.ComponentType<P>) =>
  (props: Pick<P, Exclude<keyof P, keyof ServiceConsumer>>) => <Consumer>
    {(services) => <Component services={services} {...props} />}
  </Consumer>;
