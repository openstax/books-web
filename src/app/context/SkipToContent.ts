import React, { RefObject } from 'react';

export const MAIN_CONTENT_ID = 'main-content';

const {Consumer, Provider} = React.createContext({
  registerMainContent: (_mainContent: RefObject<any>): void => {
    throw new Error('BUG: MainContent must be inside SkipToContentWrapper');
  },
});

export {
  Consumer,
  Provider,
};
