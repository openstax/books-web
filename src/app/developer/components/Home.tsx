import React from 'react';
import styled from 'styled-components';
import Layout from '../../components/Layout';
import { bodyCopyRegularStyle, H1 } from '../../components/Typography';
import { contentWrapperMaxWidth } from '../../content/components/constants';
import DisplayNotifications from '../../notifications/components/Notifications';
import Books from './Books';
import Notifications from './Notifications';
import Routes from './Routes';

// tslint:disable-next-line:variable-name
const HomeStyle = styled.div`
  ${bodyCopyRegularStyle}
  max-width: ${contentWrapperMaxWidth}rem;
  margin: 0 auto;
  padding: 2rem 0;
`;

// tslint:disable-next-line:variable-name
const Home: React.SFC = () => <Layout>
  <DisplayNotifications />
  <HomeStyle>
    <H1>REX Developer Homepage</H1>
    <Books />
    <Notifications />
    <Routes />
  </HomeStyle>
</Layout>;

export default Home;
