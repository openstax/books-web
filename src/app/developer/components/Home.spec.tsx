import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import createTestServices from '../../../test/createTestServices';
import createTestStore from '../../../test/createTestStore';
import * as Services from '../../context/Services';
import MessageProvider from '../../MessageProvider';
import Home from './Home';

describe('Home', () => {
  let services: ReturnType<typeof createTestServices>;

  beforeEach(() => {
    services = createTestServices();
  });

  it('matches snapshot', async() => {
    jest.spyOn(Date.prototype, 'getFullYear').mockReturnValue(2021);
    const store = createTestStore({navigation: new URL('https://localhost') as any});
    const component = renderer.create(<Provider store={store}>
      <Services.Provider value={services}>
        <MessageProvider>
          <Home />
        </MessageProvider>
      </Services.Provider>
    </Provider>);

    // defer promises...
    await new Promise((resolve) => setTimeout(resolve, 1));

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
