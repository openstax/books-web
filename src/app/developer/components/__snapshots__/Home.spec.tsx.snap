// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home matches snapshot 1`] = `
Array [
  .c0 {
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
  margin: 0;
  position: absolute;
  left: 0;
  top: 0;
  height: 1px;
  width: 1px;
  overflow: hidden;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:focus {
  -webkit-clip: auto;
  clip: auto;
  height: auto;
  width: auto;
  z-index: 20;
}

<a
    className="c0"
    href="#main-content"
    onClick={[Function]}
  >
    Skip to Content
  </a>,
  .c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 5rem;
  max-width: 117rem;
  margin: 0 auto;
}

.c2 {
  width: auto;
  height: 3rem;
}

.c3 {
  font-family: Neue Helvetica W01;
  color: #424242;
  font-size: 1.8rem;
  line-height: 2.5rem;
  -webkit-letter-spacing: -0.04rem;
  -moz-letter-spacing: -0.04rem;
  -ms-letter-spacing: -0.04rem;
  letter-spacing: -0.04rem;
  padding: 1rem 0 1rem 0;
  margin: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-weight: bold;
  color: #5e6062;
  padding: 1rem 0;
}

.c3:hover,
.c3:active,
.c3:focus {
  padding-bottom: 0.6rem;
  border-bottom: 0.4rem solid #63a524;
}

.c0 {
  z-index: 5;
  background: #fff;
  position: relative;
  padding: 0 3.2rem;
  box-shadow: 0 0.2rem 0.2rem 0 rgba(0,0,0,0.1);
}

@media (max-width:64em) {
  .c1 {
    height: 3.6rem;
  }
}

@media (max-width:64em) {
  .c2 {
    height: 2rem;
  }
}

@media (max-width:64em) {
  .c3 {
    font-size: 1.6rem;
    line-height: 2rem;
  }
}

@media (max-width:64em) {
  .c3 {
    padding: 0.7rem 0;
  }

  .c3:hover,
  .c3:active,
  .c3:focus {
    padding-bottom: 0.3rem;
  }
}

@media (max-width:64em) {
  .c0 {
    padding: 0 1.6rem;
  }
}

<div
    className="c0"
  >
    <div
      className="c1"
    >
      <a
        href="/"
      >
        <img
          alt="OpenStax Logo"
          className="c2"
          role="img"
          src="logo.svg"
        />
      </a>
      <a
        className="c3"
        href="https://accounts-dev.openstax.org/login"
      >
        Login
      </a>
    </div>
  </div>,
  .c1 {
  font-family: Neue Helvetica W01;
  color: #424242;
  font-size: 4.8rem;
  line-height: 5rem;
  -webkit-letter-spacing: -0.04rem;
  -moz-letter-spacing: -0.04rem;
  -ms-letter-spacing: -0.04rem;
  letter-spacing: -0.04rem;
  padding: 0 0 1rem 0;
  margin: 0;
}

.c2 {
  font-family: Neue Helvetica W01;
  color: #424242;
  font-size: 3.6rem;
  line-height: 4rem;
  -webkit-letter-spacing: -0.04rem;
  -moz-letter-spacing: -0.04rem;
  -ms-letter-spacing: -0.04rem;
  letter-spacing: -0.04rem;
  padding: 2rem 0 1rem 0;
  margin: 0;
}

.c4 {
  font-family: Neue Helvetica W01;
  color: #424242;
  font-size: 2.4rem;
  line-height: 3rem;
  -webkit-letter-spacing: -0.04rem;
  -moz-letter-spacing: -0.04rem;
  -ms-letter-spacing: -0.04rem;
  letter-spacing: -0.04rem;
  padding: 1.5rem 0 1rem 0;
  margin: 0;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-family: Neue Helvetica W01;
  border-radius: 0.2rem;
  font-size: 1.6rem;
  height: 4rem;
  padding: 0 3rem;
  color: #5e6062;
  background-color: #fff;
  border: 1px solid #d5d5d5;
  font-weight: regular;
}

.c6:hover {
  background-color: #fafafa;
}

.c6:active {
  background-color: #e5e5e5;
}

.c5 {
  color: #027EB5;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-bottom: #027EB5 solid 0.02em;
}

.c5:hover {
  color: #0064A0;
}

.c3 {
  padding: 1rem;
}

.c0 {
  font-family: Neue Helvetica W01;
  color: #424242;
  font-size: 1.6rem;
  line-height: 2.5rem;
  max-width: 117rem;
  margin: 0 auto;
  padding: 2rem 0;
}

.c0 a {
  color: #027EB5;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-bottom: #027EB5 solid 0.02em;
}

.c0 a:hover {
  color: #0064A0;
}

@media (max-width:64em) {
  .c4 {
    font-size: 1.6rem;
    line-height: 2rem;
  }
}

<div
    className="c0"
  >
    <h1
      className="c1"
    >
      REX Developer Homepage
    </h1>
    <div>
      <h2
        className="c2"
      >
        Books
      </h2>
      <div
        className="c3"
      >
        <div>
          <h3
            className="c4"
          >
            <a
              className="c5"
              href="books/book-slug-1/pages/1-test-page-2"
              onClick={[Function]}
            >
              Test Book 1
            </a>
          </h3>
        </div>
      </div>
    </div>
    <div>
      <h2
        className="c2"
      >
        Notifications
      </h2>
      <div
        className="c3"
      >
        <button
          className="c6"
          onClick={[Function]}
        >
          update available
        </button>
      </div>
    </div>
    <div>
      <h2
        className="c2"
      >
        Routes
      </h2>
      <div
        className="c3"
      >
        <div>
          <h3
            className="c4"
          >
            Developer Home
          </h3>
          path: 
          /
        </div>
        <div>
          <h3
            className="c4"
          >
            Content
          </h3>
          path: 
          /books/:book/pages/:page
        </div>
        <div>
          <h3
            className="c4"
          >
            NotFound
          </h3>
          path: 
          /(.*)
        </div>
      </div>
    </div>
  </div>,
]
`;
